app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -1,148 +1,166 @@
+    window.addEventListener('resize', this.hasResized.bind(this))
+    window.addEventListener('scroll', this.hasScrolled.bind(this))
 
-      setInterval(self.checkResize, _interval)
-      setInterval(self.checkScroll, _interval)
+    setInterval(this.checkResize.bind(this), this._interval)
+    setInterval(this.checkScroll.bind(this), this._interval)
 
-      self.$anchors = $el.find('.js-page-contents a')
-      self.getAnchors()
+    this.anchors = this.$module.querySelectorAll('.js-page-contents a')
+    this.getAnchors()
 
-      self.checkResize()
-      self.checkScroll()
-    }
+    this.checkResize()
+    this.checkScroll()
+  }
 
-    self.hasResized = function () {
-      _hasResized = true
-      return _hasResized
-    }
+  HighlightActiveSectionHeading.prototype.hasResized = function () {
+    this._hasResized = true
+    return this._hasResized
+  }
 
-    self.hasScrolled = function () {
-      _hasScrolled = true
-      return _hasScrolled
-    }
+  HighlightActiveSectionHeading.prototype.hasScrolled = function () {
+    this._hasScrolled = true
+    return this._hasScrolled
+  }
 
-    self.checkResize = function () {
-      if (_hasResized) {
-        _hasResized = false
-        _hasScrolled = true
-      }
+  HighlightActiveSectionHeading.prototype.checkResize = function () {
+    if (this._hasResized) {
+      this._hasResized = false
+      this._hasScrolled = true
     }
+  }
 
-    self.checkScroll = function () {
-      if (_hasScrolled) {
-        _hasScrolled = false
-        var windowDimensions = self.getWindowDimensions()
-        if (windowDimensions.width <= 768) {
-          self.removeActiveItem()
-        } else {
-          self.updateActiveNavItem()
-        }
+  HighlightActiveSectionHeading.prototype.checkScroll = function () {
+    if (this._hasScrolled) {
+      this._hasScrolled = false
+      var windowDimensions = this.getWindowDimensions()
+      if (windowDimensions.width <= 768) {
+        this.removeActiveItem()
+      } else {
+        this.updateActiveNavItem()
       }
     }
+  }
 
-    self.getAnchors = function () {
-      $.each(self.$anchors, function (i) {
-        var anchorID = $(this).attr('href')
-        // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
-        anchorIDs.push(anchorID)
-      })
+  HighlightActiveSectionHeading.prototype.getWindowDimensions = function () {
+    return {
+      height: window.innerHeight,
+      width: window.innerWidth
     }
+  }
 
-    self.getHeadingPosition = function ($theID) {
-      return $theID.offset()
+  HighlightActiveSectionHeading.prototype.getAnchors = function () {
+    for (var i = 0; i < this.anchors.length; i++) {
+      var anchorID = this.anchors[i].getAttribute('href')
+      // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
+      this.anchorIDs.push(anchorID)
```

Comment:
This has my spider senses tingling with the assumption that all href attributes are anchor links 🤔 

Can see it's in the past code though.

Reply:
Oh sure, if we can drop it that'd be a good outcome. I do worry about the maintenance anything in here gets.